dcm4che-tool/dcm4che-tool-dcmdir/src/main/java/org/dcm4che3/tool/dcmdir/DcmDir.java

```diff
@@ -230,7 +269,48 @@ public static void main(String[] args) {
         }
     }
 
-    public void compact(File f, File bak) throws IOException {
+    private int readCSVFile(CommandLine cl, DcmDir main, int num) throws Exception {
+        if (!cl.hasOption("record-config"))
+            throw new MissingOptionException("Missing record-config option.");
+        String delim = cl.hasOption("csv-delim") ? cl.getOptionValue("csv-delim") : ",";
+        String quote = cl.hasOption("csv-quote") && !cl.getOptionValue("csv-quote").equals("")
+                        ? cl.getOptionValue("csv-quote") : "\"";
+        BufferedReader br = null;
+        try {
+            br = new BufferedReader(new FileReader(cl.getOptionValue("csv")));
+            loadDefaultConfiguration(cl.getOptionValue("record-config"));
+            String headerline = br.readLine();
+            String[] headers = headerline.split(delim);
+            List<Integer> tags = new ArrayList<Integer>();
+            List<VR> vrs = new ArrayList<VR>();
+            for (int i = 0; i < headers.length; i++) {
+                String header = headers[i].replaceAll(quote, "");
+                int tag = DICT.tagForKeyword(header);
+                tags.add(tag);
+                vrs.add(DICT.vrOf(tag));
+            }
+
+            String nextLine;
+            while((nextLine = br.readLine()) != null) {
+                String escq = new StringBuilder().append("\\").append(quote).toString();
```

Comment:
Performance: Build this string only once before the while loop. Same for the next line. (String regex).

(By the way: Why use a StringBuilder here, but not for the next line? I wouldn't use a StringBuilder at all in this particular case. Java automatically replaces such a type of string concatenation by a StringBuilder in the compiled bytecode.)